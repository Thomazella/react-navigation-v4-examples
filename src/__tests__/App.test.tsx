/**
 * @format
 */

import 'react-native';
import React from 'react';
import { View, Text } from 'react-native';
import { createAppContainer } from 'react-navigation';
import { createStackNavigator } from 'react-navigation-stack';
import { render, toJSON } from '@testing-library/react-native';

import App from '../App';

describe('<App />', () => {
  it('should render properly', () => {
    const { baseElement } = render(<App />);
    expect(toJSON(baseElement)).toMatchSnapshot();
  });
});

const TestScreen = () => {
  return (
    <View>
      <Text>Test Screen</Text>
    </View>
  );
};

describe('StackNavigator', () => {
  it('should render properly', () => {
    const Navigator = createStackNavigator({
      Home: TestScreen,
    });
    const Container = createAppContainer(Navigator);
    const { baseElement } = render(<Container />);
    expect(toJSON(baseElement)).toMatchSnapshot();
  });
});

it('renders correctly', () => {
  renderer.create(<App />);
});
