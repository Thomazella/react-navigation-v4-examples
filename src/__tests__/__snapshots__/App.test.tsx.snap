// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render properly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView>
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={
          Object {
            "backgroundColor": "#F3F3F3",
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            Object {
              "backgroundColor": "#F3F3F3",
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            }
          }
        >
          <Image
            accessibilityRole="image"
            source={
              Object {
                "testUri": "../Libraries/NewAppScreen/components/logo.png",
              }
            }
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": undefined,
                  "width": undefined,
                },
                Object {
                  "marginBottom": -192,
                  "marginLeft": -128,
                  "opacity": 0.2,
                  "overflow": "visible",
                  "resizeMode": "cover",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 40,
                "fontWeight": "600",
                "textAlign": "center",
              }
            }
          >
            Welcome to React
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFF",
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Step One
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                }
              }
            >
              Edit 
              <Text
                style={
                  Object {
                    "fontWeight": "700",
                  }
                }
              >
                App.tsx
              </Text>
               to change this screen and then come back to see your edits.
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              See Your Changes
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                }
              }
            >
              <Text>
                Press 
                <Text
                  style={
                    Object {
                      "fontWeight": "700",
                    }
                  }
                >
                  Cmd + R
                </Text>
                 in the simulator to reload your app's code.
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Debug
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                }
              }
            >
              <Text>
                Press 
                <Text
                  style={
                    Object {
                      "fontWeight": "700",
                    }
                  }
                >
                  Cmd + D
                </Text>
                 in the simulator or
                 
                <Text
                  style={
                    Object {
                      "fontWeight": "700",
                    }
                  }
                >
                  Shake
                </Text>
                 your device to open the React Native debug menu.
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                  "fontWeight": "600",
                }
              }
            >
              Learn More
            </Text>
            <Text
              style={
                Object {
                  "color": "#444",
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                }
              }
            >
              Read the docs to discover what to do next:
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 32,
                "paddingHorizontal": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                The Basics
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                Explains a Hello World for React Native.
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                Style
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                Covers how to use the prop named style which controls the visuals.
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                Layout
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                React Native uses flexbox for layout, learn how it works.
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                Components
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                The full list of components and APIs inside React Native.
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                Navigation
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                How to handle moving between screens inside your application.
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                Networking
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                How to use the Fetch API in React Native.
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                Help
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                Need more help? There are many other React Native developers who may have the answer.
              </Text>
            </TouchableOpacity>
            <View
              style={
                Object {
                  "backgroundColor": "#DAE1E7",
                  "height": 1,
                }
              }
            />
            <TouchableOpacity
              accessibilityRole="button"
              activeOpacity={0.2}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "justifyContent": "space-between",
                  "paddingVertical": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#1292B4",
                    "flex": 2,
                    "fontSize": 18,
                    "fontWeight": "400",
                  }
                }
              >
                Follow us on Twitter
              </Text>
              <Text
                style={
                  Object {
                    "color": "#444",
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  }
                }
              >
                Stay in touch with the community, join in on Q&As and more by following React Native on Twitter.
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  </View>
</View>
`;

exports[`StackNavigator should render properly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNCSafeAreaView
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              pointerEvents="box-none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              transitionSpec={
                Object {
                  "close": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 0.01,
                      "restSpeedThreshold": 0.01,
                      "stiffness": 1000,
                    },
                  },
                  "open": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 0.01,
                      "restSpeedThreshold": 0.01,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "flex": 1,
                    "marginTop": 64,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "flex": 1,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 1,
                          "width": -1,
                        },
                        "shadowOpacity": 0.3,
                        "shadowRadius": 5,
                        "top": 0,
                        "width": 3,
                      }
                    }
                  />
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          undefined,
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <View>
                          <Text>
                            Test Screen
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            pointerEvents="box-none"
            style={null}
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "borderBottomColor": "#a7a7aa",
                    "flex": 1,
                    "shadowColor": "#a7a7aa",
                    "shadowOffset": Object {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.85,
                    "shadowRadius": 0,
                  }
                }
              />
            </View>
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "height": 64,
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "opacity": undefined,
                  "transform": undefined,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "height": 20,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Object {
                      "marginHorizontal": 18,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityRole="header"
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.9)",
                        "fontSize": 17,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Home
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNCSafeAreaView>
    </View>
  </View>
</View>
`;
